### PR TITLE
Protect from disasters while gather_stats

### PR DESCRIPTION
**Pull request by mroda88**
Add protections against failures in monitoring thread.

### CODE REVIEW

File: src/detail/DAQModuleManager.hxx [hunk 1]
```diff
@@ -225,7 +225,18 @@ DAQModuleManager::gather_stats(opmonlib::InfoCollector& ci, int level)
...
+    catch( ers::Issue & i ) {
+      ers::error( FailedInfoGathering(ERS_HERE, mod_name, i) );
+    }
+    catch( std::exception & ex ) {
+      ers::error( ExceptionWhileInfoGathering(ERS_HERE, mod_name, ex.what()) );
+    }
+    catch( ... ) {
+      ers::error( FailedInfoGathering(ERS_HERE, mod_name) );
+    }
     if (!tmp_ci.is_empty()) {
```
Comment (alessandrothea) [inline src/detail/DAQModuleManager.hxx]: I suspect that if an exception is thrown `tmp_ci` may not be in an healthy state

Comment (alessandrothea) [reply src/detail/DAQModuleManager.hxx]: And with that I meant  : in a well know state. Maybe better to move the if statement inside the try catch

File: src/detail/DAQModuleManager.hxx [hunk 2]
```diff
@@ -225,7 +225,18 @@ DAQModuleManager::gather_stats(opmonlib::InfoCollector& ci, int level)
 
   for (const auto& [mod_name, mod_ptr] : m_module_map) {
     opmonlib::InfoCollector tmp_ci;
-    mod_ptr->get_info(tmp_ci, level);
+    try {
+      mod_ptr->get_info(tmp_ci, level);
+    }
+    catch( ers::Issue & i ) {
+      ers::error( FailedInfoGathering(ERS_HERE, mod_name, i) );
```
Comment (alessandrothea) [inline src/detail/DAQModuleManager.hxx]: Error or warning? I would go with warning maybe

**Review by alessandrothea — CHANGES_REQUESTED** [review]

**Review by alessandrothea — APPROVED** [review]

**Review by bieryAtFnal — APPROVED** [review]
LGTM2 (regression testing)

**Pull request merged**